Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift

```diff
@@ -32,13 +32,26 @@ public protocol NavigationServiceDelegate: AnyObject, UnimplementedLogging {
     /**
      Called immediately before the navigation service calculates a new route.
      
-     This method is called after `navigationService(_:shouldRerouteFrom:)` is called, simultaneously with the `Notification.Name.routeControllerWillReroute` notification being posted, and before `navigationService(_:didRerouteAlong:)` is called.
+     This method is called after `navigationService(_:shouldRerouteFrom:)` is called, simultaneously with the `Notification.Name.routeControllerWillReroute` notification being posted, and before `navigationService(_:willModify:)` is called.
      
      - parameter service: The navigation service that will calculate a new route.
      - parameter location: The user’s current location.
      */
     func navigationService(_ service: NavigationService, willRerouteFrom location: CLLocation)
     
+    /**
+     When reroute is happening, navigation service suggests to customize the `RouteOptions` used to calculate new route.
+     
+     This method is called after `navigationService(_:willRerouteFrom:)` is called, and before `navigationService(_:didRerouteAlong:)` is called. This method is not called on proactive rerouting.
+     
+     Default implementation does no modifications.
+     
+     - parameter service: The navigation service that will calculate a new route.
+     - parameter options: Original `RouteOptions`.
+     - returns: Modified `RouteOptions`.
+     */
+    func navigationService(_ service: NavigationService, willModify options: RouteOptions) -> RouteOptions
```

Comment:
I don't know if it will be clear to customers what this delegate method is doing.
Should we rename it to something like `optionsForReroute:`?

Reply:
I wanted to emphasise that options should be modified in this method... But something like `modifiedOptionsForReroute` sounds too bulky.

Reply:
But on the other hand we have `shouldPreventReroutesWhenArrivingAt` so previous variant doesn't seem too much :)